### PR TITLE
Use official docker image

### PR DESCRIPTION
**Pull request by rail**

### CODE REVIEW

File: .taskcluster.yml [hunk 1]
```diff
@@ -18,21 +18,20 @@ tasks:
 
     owner: ${event.sender.login}@users.noreply.github.com
 
-    shouldPush:
+    push_docker_image:
       $if: 'tasks_for == "github-pull-request"'
-      then: false
+      then: '0'
```
Comment (MihaiTabara) [inline .taskcluster.yml]: Coming from reviewing a bash script, I almost said this is reversed :)

File: .taskcluster.yml [hunk 2]
```diff
@@ -75,88 +74,58 @@ tasks:
...
+      workerType: linux
+      created: {$fromNow: ''}
+      deadline: {$fromNow: '24 hours'}
+      payload:
+        features:
+          # Needed for access to secret
+          taskclusterProxy: true
+          dind: true
+        maxRunTime: 3600
+        image: 'docker:1.6.2'
```
Comment (MihaiTabara) [inline .taskcluster.yml]: 👍

File: .taskcluster.yml [hunk 3]
```diff
@@ -75,88 +74,58 @@ tasks:
...
-            - bash
-            - -ce
-            - >-
-              git clone ${repository} /src &&
-              cd /src &&
-              git config advice.detachedHead false &&
-              git checkout ${head_rev} &&
-              docker.d/generate_version_json.sh &&
-              docker build -f Dockerfile -t mozilla/releng-k8s-autoscale:${docker_tag} . &&
-              docker save mozilla/releng-k8s-autoscale:${docker_tag} > /tmp/image.tar &&
```
Comment (MihaiTabara) [inline .taskcluster.yml]: I like that we're no longer saving it locally. Saves time and space! 👍

File: .taskcluster.yml [hunk 4]
```diff
@@ -75,88 +74,58 @@ tasks:
...
-              type: file
-        scopes:
-          - secrets:get:project/releng/k8s-autoscale/deploy
-        metadata:
-          name: build and push docker image (${docker_tag})
-          description: build and push docker image (${docker_tag})
-          owner: ${owner}
-          source: ${repository}/raw/${head_rev}/.taskcluster.yml
-
-      else: # run docker build to verify it works
```
Comment (MihaiTabara) [inline .taskcluster.yml]: Much cleaner to bake into an if/else just the push operation. Looks cleaner, thank you

File: docker.d/push_image.sh [hunk 5]
```diff
@@ -1,11 +1,16 @@
-#!/bin/bash
+#!/bin/sh
 set -e
 test $SECRET_URL
 test $DOCKERHUB_EMAIL
 test $DOCKERHUB_USER
 test $TAG
 
-dockerhub_password=$(curl $SECRET_URL | python -c 'import json, sys; a = json.load(sys.stdin); print a["secret"]["docker"]["password"]')
+apk -U add jq
```
[the remark this answers is not in the document]
Comment (rail) [reply docker.d/push_image.sh]: I reject 100% certainty :D It worked fine before, should run in the nearest future. :)

File: .taskcluster.yml (hunk 1 above)
Comment (rail) [reply .taskcluster.yml]: I've just unified this with the approach we took in scriptworker-scripts.

File: docker.d/push_image.sh [hunk 6]
```diff
@@ -1,11 +1,16 @@
...
+
+dockerhub_password=$(wget -qO- $SECRET_URL | jq '.["secret"]["docker"]["password"]')
```
[the remark this answers is not in the document]
Comment (rail) [reply docker.d/push_image.sh]: We could've written this script in Haskell and compile it every time in CI. :P

**Pull request merged**